hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleAccountState.java

```diff
@@ -75,6 +82,9 @@
 	private ByteString alias = DEFAULT_ALIAS;
 	private int autoAssociationMetadata;
 	private int numContractKvPairs;
+	private Map<EntityNum, Long> cryptoAllowances = new HashMap<>();
```

Comment:
Hmm...I think we need to make `EntityNum implement Comparable<EntityNum>` and then either:
1. Use a `TreeMap` here instead of `HashMap`; or
2. Change `serialize()` to iterate through the map keys in sorted order while writing.

Otherwise, a reconnecting node can get an ISS because its `HashMap` may resolve collisions differently than nodes who handled every transaction---and thus serialize the map keys in a different order!

Reply:
Agreed. Thanks @tinker-michaelj ! 
Modified to implement `Comparable` for `EntityNum` and `FcAllowanceId`  and use `TreeMap` instead of `HashMap`.